### DISCUSSION TITLE
$compute clause

### DISCUSSION THREAD

**Discussion by trojanmartin** [AutoMapper/AutoMapper.Extensions.OData#115] [Ideas]
I'm thinking about implementation of the $compute OdataQueryOption. In current approach this library is using automapper ProjectTo method which does the projection. With $compute is possible add dynamic properties. Any advices how that can be achieved with current approach?

**Comment by BlaiseD** [comment]
It looks very involved.  Take the filter example from [dev-blogs here](https://devblogs.microsoft.com/odata/compute-and-search-in-asp-net-core-odata-8/#compute-in-filter).

Assuming:
```c#
    public class MyModel
    {
        double Price { get; set; }
        double TotalPrice { get; set; }
        int Qty { get; set; }
    }

    public class MyData
    {
        double Price { get; set; }
        int Qty { get; set; }
    }
```

- First step: create the `TotalPrice` expression from the `Price` expression.
- Second step: map or project.
- Third step: call the database.

The first step happens irrespective of `Map` or `Project`.  Then considering that the `$compute` expression can be multiple things I'll anticipate writing some code like [here in the FilterHelper](https://github.com/AutoMapper/AutoMapper.Extensions.OData/blob/1a19647bb351f2889a5706cbe1f8bd72e6fde378/AutoMapper.AspNetCore.OData.EFCore/FilterHelper.cs#L43).

Plus different solutions for different `$compute` use cases.

You'll be able to tell us more if you decide to get into it.

Thanks.
